### PR TITLE
Directly wire ctx into http request

### PR DESCRIPTION
**Pull request by rht**

### CODE REVIEW

File: commands/http/client.go [hunk 1]
```diff
@@ -101,46 +94,28 @@ func (c *client) Send(req cmds.Request) (cmds.Response, error) {
 	}
 	httpReq.Header.Set(uaHeader, config.ApiVersion)
 
-	ec := make(chan error, 1)
-	rc := make(chan cmds.Response, 1)
-	dc := req.Context().Done()
+	httpReq.Cancel = req.Context().Done()
+	httpReq.Close = true
```
Comment (whyrusleeping) [inline commands/http/client.go]: what happens if we don't set close?

Comment (rht) [inline commands/http/client.go]:
```
	// Close indicates whether to close the connection after
	// replying to this request (for servers) or after sending
	// the request (for clients).
```[1]

ref:
[1] https://golang.org/pkg/net/http/#Request

Comment (whyrusleeping) [inline commands/http/client.go]: I didnt mean literally, i meant 'does our shit stop working properly' ?

Comment (rht) [inline commands/http/client.go]: No, it's just that dangling transports are costly.

Comment (whyrusleeping) [inline commands/http/client.go]: the dangling transports *should* be reused by the library though, thats the whole point of it afaict

Comment (rht) [inline commands/http/client.go]: But there is not explicit close at elsewhere in the code. Either kill it after one request, or set a gc (which remains to be implemented).

**Comment by whyrusleeping** [comment]
alright, LGTM

**Pull request merged**